week8/src/Main.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * Homework Week 8
+ *
+ * @author Florin
+ */
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.*;
+
+public class Main {
+    public static void main(String[] args) throws IOException {
+
+        List <Student> sList = new ArrayList<>();
+        sList.add(new Student("Walter", "White", 1968, "Male", 2675));
+        sList.add(new Student("Jessie", "Pinkman", 1990, "Male", 2839));
+        sList.add(new Student("Saul", "Goodman", 1975, "Male", 2413));
+        sList.add(new Student("Mike", "Ehrmantraut", 1958, "Male", 3293));
+        sList.add(new Student("Hank", "Schrader", 1968, "Male", 2802));
+
+        List <Student> byAge = new ArrayList<>();
+
+        Scanner sc = new Scanner(System.in);
+
+        int choice;
+
+        do {
+            System.out.println("[1] Add student");
+            System.out.println("[2] Delete student");
+            System.out.println("[3] Retrieve all students by age");
+            System.out.println("[4] Exit");
+
+            int tempChoice = sc.nextInt();
+            choice = tempChoice;
+
+            switch (choice) {
+
+                //Add student
+                case 1:
+                    String name = name();
+                    String surname = surname();
+                    int year = birthYear();
+                    String gender = genderSelect();
+                    int id = generateId();
+                    sList.add(new Student(name,surname,year,gender,id));
+                    System.out.println("Student successfully added! Press [Enter] to continue...");
+                    try{System.in.read();}
+                    catch(Exception e){}
+                    break;
+
+                //Remove student by ID
+                case 2:
+                    sList.forEach(System.out::println);
+                    int x = removeStudent();
+                    for (int i = 0; i < sList.size(); i++) {
+                        if (sList.get(i).getId() == x){
+                            sList.remove(i);
+                            System.out.println("Student successfully removed! Press [Enter] to continue...");
+                        }
+                    }
```

Comment:
We shouldn't remove objects from a list while iterating using a for-loop. We might get a `ConcurrentModificationException`.
Instead, we should use `Iterator`'s `remove()` method.